go/sqltypes/arithmetic_test.go

```diff
@@ -67,21 +170,15 @@ func TestAdd(t *testing.T) {
 		// Make sure underlying error is returned while adding.
 		v1:  NewInt64(-1),
 		v2:  NewUint64(2),
-		err: vterrors.New(vtrpcpb.Code_INVALID_ARGUMENT, "cannot add a negative number to an unsigned integer: 2, -1"),
+		out: NewInt64(-9223372036854775808),
 	}, {
 		// Make sure underlying error is returned while converting.
 		v1:  NewFloat64(1),
 		v2:  NewFloat64(2),
-		err: vterrors.New(vtrpcpb.Code_INVALID_ARGUMENT, "unexpected type conversion: FLOAT64 to INT64"),
+		out: NewInt64(3),
```

Comment:
need a test for something like `1+'a'`, to show that it won't fail, and one for the other test. I just noticed this, which means `Add` should also not return an error. It should treat unconvertable strings as 0.
```
mysql> create table a(id int, val varchar(129), primary key(id));
Query OK, 0 rows affected (0.03 sec)

mysql> insert into a values(1,'aa'), (2,'2');
Query OK, 2 rows affected (0.01 sec)
Records: 2  Duplicates: 0  Warnings: 0

mysql> select id+val from a;
+--------+
| id+val |
+--------+
|      1 |
|      4 |
+--------+
2 rows in set (0.00 sec)
```